powerstore/resource_host_group.go

```diff
@@ -0,0 +1,249 @@
+package powerstore
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	client "terraform-provider-powerstore/client"
+	"terraform-provider-powerstore/models"
+
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// newHostGroupResource returns host group new resource instance
+func newHostGroupResource() resource.Resource {
+	return &resourceHostGroup{}
+}
+
+type resourceHostGroup struct {
+	client *client.Client
+}
+
+// Metadata defines resource interface Metadata method
+func (r *resourceHostGroup) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_hostgroup"
+}
+
+// Schema defines resource interface Schema method
+func (r *resourceHostGroup) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "HostGroup resource",
+
+		Attributes: map[string]schema.Attribute{
+
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Unique identifier of the host group.",
+				MarkdownDescription: "Unique identifier of the host group.",
+			},
+
+			"name": schema.StringAttribute{
+				Required:            true,
+				Description:         "The host group name.",
+				MarkdownDescription: "The host group name.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"description": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "An optional description for the host group.",
+				MarkdownDescription: "An optional description for the host group.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"host_ids": schema.SetAttribute{
```

Comment:
Is it expected to support host names as well?

Reply:
not the requirement in this PR. will address through another PR if required.

Reply:
For most of the resources, we are supporting names and ids. Is it not implicit that host_name should be supported? Please correct me if I am wrong.

Reply:
As discussed, will address this in different PR.